src/tests/LoggingTest.cpp

```diff
@@ -95,18 +98,33 @@ void LoggingTest::envVarTest() {
   std::ostringstream out;
   Cr::Utility::Debug debugCapture{&out};
   Cr::Utility::Warning warnCapture{&out};
+  // use contains to bypass issue with timestamp
```

Comment:
Though ... the test code you have now won't catch if the logger prints more than it should :thinking: 

Since the timestamp has a fixed-width formatting, what if you'd, instead of `contains()`, compare just the suffix after?

```cpp
  CORRADE_COMPARE(Cr::Containers::StringView{out.str()}.suffix(42),
                  Cr::Containers::StringView{data.defaultDebug});
```

etc., where `42` is the actual size of the timestamp prefix? (Lazy to count, sorry :P)

Reply:
Some of them are empty though, so the suffix pukes.

Reply:
I could compare with a \n

Reply:
oh right, then ... a branch?

```cpp
  Cr::Containers::StringView actual = out.str();
  CORRADE_COMPARE(actual.suffix(actual.size() < 42 : 0 : 42),
                  Cr::Containers::StringView{data.defaultDebug});
```

A less hardcoded variant but with a worse diagnostic would be using `hasSuffix()` instead of `contains()`.

(Yes, goes straight to my TODOs, this desperately needs some "String has a suffix" comparator.)

Reply:
Looks like I got it by appending newlines, which are expected.